src/app/rundown-view/services/mini-shelf-cycle.service.spec.ts

```diff
@@ -0,0 +1,149 @@
+  })
+
+  it('should set as next the first MiniShelf after a segment onAir on cycle forwards and not the second one', () => {
+    const actionService: ActionService = mock<ActionService>()
+    when(actionService.executeAction(anyString(), anyString())).thenReturn(instance(mock<Observable<void>>()) as Observable<void>)
+
+    const actionStateService: ActionStateService = mock<ActionStateService>()
+    when(actionStateService.getRundownActions(anyString())).thenReturn([testEntityFactory.createTv2VideoClipAction('firstMiniShelf'), testEntityFactory.createTv2VideoClipAction('secondMiniShelf')])
+
+    const segment: Segment = testEntityFactory.createSegment({
+      id: 'segmentOnAir',
+      isOnAir: true,
+      isNext: true,
+      isHidden: false,
+      parts: [
+        testEntityFactory.createPart({
+          isOnAir: true,
+          metadata: { actionId: 'testPartOnAir' },
+        }),
+        testEntityFactory.createPart({
+          isNext: true,
+          metadata: { actionId: 'testPartNext' },
+        }),
+      ],
+    })
+
+    const firstMiniShelf: Segment = testEntityFactory.createMiniShelfSegment({
+      id: 'firstMiniShelf',
+      miniShelfVideoClipFile: 'firstMiniShelf',
+    })
+    const secondMiniShelf: Segment = testEntityFactory.createMiniShelfSegment({
+      id: 'secondMiniShelf',
+      miniShelfVideoClipFile: 'secondMiniShelf',
+    })
+
+    const miniShelfNavigationService: MiniShelfNavigationService = mock<MiniShelfNavigationService>()
+    when(miniShelfNavigationService.getPreviousMiniShelfSegment(anything(), anything())).thenReturn(firstMiniShelf)
+
+    const testee: MiniShelfCycleService = createTestee({
+      actionService: instance(actionService),
+      actionStateService: instance(actionStateService),
+      miniShelfNavigationService: instance(miniShelfNavigationService),
+    })
+
+    testee.cycleMiniShelfBackward(
+      testEntityFactory.createRundown({
+        id: 'rundown-id',
+        isActive: true,
+        segments: [segment, firstMiniShelf, secondMiniShelf],
+      })
+    )
+
+    verify(actionService.executeAction('actionId_firstMiniShelf', 'rundown-id')).once()
+
+    verify(actionService.executeAction('actionId_secondMiniShelf', 'rundown-id')).never()
+  })
+})
+
+function createTestee(params?: { actionService?: ActionService; miniShelfNavigationService?: MiniShelfNavigationService; actionStateService?: ActionStateService }): MiniShelfCycleService {
+  let miniShelfNavigationService: MiniShelfNavigationService
+  if (!params?.miniShelfNavigationService) {
+    const miniShelfNavigationServiceMock: MiniShelfNavigationService = mock<MiniShelfNavigationService>()
+    miniShelfNavigationService = instance(miniShelfNavigationServiceMock)
+  }
+
+  let actionStateService: ActionStateService
+  if (!params?.actionStateService) {
+    const actionStateServiceMock: ActionStateService = mock<ActionStateService>()
+    actionStateService = instance(actionStateServiceMock)
+  }
```

Comment:
This can be simplified to:
```
const miniShelfNavigationService: MiniShelfNavigationService = params?.miniShelfNavigationService ?? instance(mock<MiniShelfNavigationService>())
const actionStateService: ActionStateService = params?.actionStateService ?? instance(mock<ActionStateService>())

...

return new MiniShelfCycleService(miniShelfNavigationService, actionStateService, params?.actionService ?? actionService)
```